localstack/services/apigateway/templates.py

```diff
@@ -320,35 +322,61 @@ def render(self, api_context: ApiInvocationContext, **kwargs) -> Union[bytes, st
         response.status_code = int(selected_integration_response.get("statusCode", 200))
         response_templates = selected_integration_response.get("responseTemplates", {})
 
-        # we only support JSON templates for now - if there is no template we return
-        # the response as is
-        # TODO - support other content types, besides application/json (based on `Accept` request header)
+        # we only support JSON and XML templates for now - if there is no template we return the response as is
+        # If the content type is not supported we always use application/json as default value
+        # TODO - support other content types, besides application/json and application/xml
         # see https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#selecting-mapping-templates
-        template = response_templates.get(APPLICATION_JSON, {})
-        if not template:
+        accept = api_context.headers.get("accept", APPLICATION_JSON)
+        supported_types = [APPLICATION_JSON, APPLICATION_XML]
+        media_type = accept if accept in supported_types else APPLICATION_JSON
+        if media_type not in response_templates:
             return response._content
 
+        template = response_templates.get(media_type, {})
```

Comment:
nit: we could have kept the previous logic
```suggestion
        if not (template := response_templates.get(media_type, {})):
            return response._content
```

Reply:
Fixed in 4219f062